WalletWasabi/JsonConverters/UnblindedSignatureJsonConverter.cs

```diff
@@ -20,19 +20,30 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 		{
 			JArray arr = JArray.Load(reader);
 
-			string c = arr[0].Value<string>();
-			string s = arr[1].Value<string>();
+			var cs = BigInteger.Parse(arr[0].Value<string>());
+			var ss = BigInteger.Parse(arr[1].Value<string>());
 
-			return new UnblindedSignature(new BigInteger(c), new BigInteger(s));
+			var carr = cs.ToByteArray(true, true);
+			var sarr = ss.ToByteArray(true, true);
+
+			carr = new byte[32 - carr.Length].Concat(carr).ToArray();
```

Comment:
in 99.9999999% of times, `carr.Length` is 32. I think it is wastefull to create a new byte array everytimes like this.
On top of it, I think you should properly handle the case where `carr` is more than 32 bytes. (somebody sending you shit)

Reply:
That's true, 99.99999% of times it is 32 and we don't need to create a byte array in those cases. I will change that. 
If someone sends us shit this code will fail with an exception, I think we could throw a better exception instead.